### ISSUE TITLE
Installation Failing, Failure Seems to Occur at Makefile line 6708

### ISSUE THREAD

**Issue by imk1** [ImageMagick/ImageMagick#1550]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

I am running make and getting the following error:
coders/png.c: In function 'ReadOnePNGImage':
coders/png.c:2447:8: warning: unused variable 'option' [-Wunused-variable]
       *option;
        ^~~~~~
  CC       filters/MagickCore_libMagickCore_7_Q16HDRI_la-analyze.lo
  CCLD     MagickCore/libMagickCore-7.Q16HDRI.la
libtool: warning: library '/home/ikaplow/anaconda2/pkgs/freetype-2.9.1-h8a8886c_1/lib/libfreetype.la' was moved.
/bin/grep: /opt/anaconda1anaconda2anaconda3/lib/libpng16.la: No such file or directory
/bin/sed: can't read /opt/anaconda1anaconda2anaconda3/lib/libpng16.la: No such file or directory
libtool:   error: '/opt/anaconda1anaconda2anaconda3/lib/libpng16.la' is not a valid libtool archive
make[1]: *** [Makefile:6708: MagickCore/libMagickCore-7.Q16HDRI.la] Error 1
make[1]: Leaving directory '/home/ikaplow/RegulatoryElementEvolutionProject/src/ImageMagick'
make: *** [Makefile:5766: all] Error 2

### Steps to Reproduce

I ran configure in the following way:
./configure CPPFLAGS='-I/[DIRECTORY WITH ft2build.h] -I[DIRECTORY WITH zlib.h] -I[DIRECTORY WITH fontconfig/fontconfig.h] -I[DIRECTORY WITH libtool include]' LDFLAGS='-L[directory with freetype2 library] -L[directory with libpng library] -L[directory with fontconfig library] -L[directory with zlib library] -L[directory with libtool library]'
Most of my dependencies, including freetype, zlib, libtool, libpng, and fontconfig, have been installed through anaconda2.

### System Configuration

I am on a Linux cluster with Centos 6, and I do not have root access.  I install most things locally through anaconda.

- ImageMagick version: 7.0.8
- Environment (Operating system, version and so on): Linux cluster with Centos 6
- Additional information: I do not have root access.

**Comment by urban-warrior** [comment]
We updated the autoconf/automake configuration failes.  Download the latest ImageMagick release.  Does it still fail to build?

**Comment by imk1** [comment]
It still fails to build, but I get a different error:

MagickCore/utility.c: In function 'MagickDelay':
MagickCore/utility.c:1741:3: error: #error "Time delay method not defined."
 # error "Time delay method not defined."
   ^~~~~
make[1]: *** [Makefile:8738: MagickCore/libMagickCore_7_Q16HDRI_la-utility.lo] Error 1

**Comment by iamhgdabc** [comment]
> It still fails to build, but I get a different error:
> 
> MagickCore/utility.c: In function 'MagickDelay':
> MagickCore/utility.c:1741:3: error: #error "Time delay method not defined."
> 
> # error "Time delay method not defined."
> ^~~~~
> make[1]: *** [Makefile:8738: MagickCore/libMagickCore_7_Q16HDRI_la-utility.lo] Error 1

do you fix this problem， i have the same problem with you

**Comment by Abdelhady** [comment]
I still have this issue with the latest version (7.0.8-68) on Ubuntu 16.04

**Comment by urban-warrior** [comment]
Post the exception message you received from the compiler so we can determine the source of the problem.

**Comment by Abdelhady** [comment]
Well, I'm trying to install it like the following:
```bash
wget http://www.imagemagick.org/download/ImageMagick.tar.gz
cd ImageMagick-7.0.8-68
./configure --with-pango --with-webp
make         #I'm seeing the exception after running this
make install
```

And here is the full exception:
```
coders/png.c: In function ‘ReadOnePNGImage’:
coders/png.c:2449:8: warning: unused variable ‘option’ [-Wunused-variable]
       *option;
        ^
  CC       coders/MagickCore_libMagickCore_7_Q16HDRI_la-ept.lo
  CC       coders/MagickCore_libMagickCore_7_Q16HDRI_la-tiff.lo
  CC       coders/MagickCore_libMagickCore_7_Q16HDRI_la-webp.lo
coders/webp.c: In function ‘WriteAnimatedWEBPImage’:
coders/webp.c:761:3: error: unknown type name ‘WebPAnimEncoderOptions’
   WebPAnimEncoderOptions
   ^
coders/webp.c:765:3: error: unknown type name ‘WebPAnimEncoder’
   WebPAnimEncoder
   ^
coders/webp.c:775:3: warning: implicit declaration of function ‘WebPAnimEncoderOptionsInit’ [-Wimplicit-function-declaration]
   WebPAnimEncoderOptionsInit(&enc_options);
   ^
coders/webp.c:777:16: error: request for member ‘verbose’ in something not a structure or union
     enc_options.verbose = 1;
                ^
coders/webp.c:781:9: warning: implicit declaration of function ‘WebPAnimEncoderNew’ [-Wimplicit-function-declaration]
   enc = WebPAnimEncoderNew(image->page.width, image->page.height, &enc_options);
         ^
coders/webp.c:781:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   enc = WebPAnimEncoderNew(image->page.width, image->page.height, &enc_options);
       ^
coders/webp.c:800:5: warning: implicit declaration of function ‘WebPAnimEncoderAdd’ [-Wimplicit-function-declaration]
     WebPAnimEncoderAdd(enc, &picture, frame_timestamp, configure);
     ^
coders/webp.c:809:3: warning: implicit declaration of function ‘WebPAnimEncoderAssemble’ [-Wimplicit-function-declaration]
   WebPAnimEncoderAssemble(enc, &webp_data);
   ^
coders/webp.c:810:3: warning: implicit declaration of function ‘WebPMemoryWriterClear’ [-Wimplicit-function-declaration]
   WebPMemoryWriterClear(writer_info);
   ^
coders/webp.c:813:3: warning: implicit declaration of function ‘WebPAnimEncoderDelete’ [-Wimplicit-function-declaration]
   WebPAnimEncoderDelete(enc);
   ^
Makefile:9924: recipe for target 'coders/MagickCore_libMagickCore_7_Q16HDRI_la-webp.lo' failed
make[1]: *** [coders/MagickCore_libMagickCore_7_Q16HDRI_la-webp.lo] Error 1
make[1]: Leaving directory '/tmp/ImageMagick-7.0.8-68'
Makefile:5990: recipe for target 'all' failed
make: *** [all] Error 2
```

Actually yesterday I was running through Ansible, so I've seen only the first 3 lines of the exception (kind of), but today I've run it manually, & it looks clear now that the issue is in the WebP part

**Comment by urban-warrior** [comment]
Upgrade your release of WebP to 0.4.4 or above or do not add the --with-webp option to your configure script command-line.

**Comment by Abdelhady** [comment]
My current WebP version is `0.4.4-1` (which is the newest version already), where I'm installing it with `apt install webp`, 

& I do need the `--with-webp` option in my project. Actually WebP was working fine with me when ImageMagick was at version `7.0.8-33` and earlier ones.

**Comment by urban-warrior** [comment]
WebPAnimEncoderOptions should be defined in the webp/mux.h header file.  On our system, it says:

`typedef struct WebPAnimEncoderOptions WebPAnimEncoderOptions;`

Is this symbol defined on your system?

**Comment by Abdelhady** [comment]
**tl;dr:** I don't have it in my system

I'm just installing ImageMagick on a DigitalOcean droplet with a naked "Ubuntu 16.04", where I have the file at `/usr/include/webp/mux.h`, and here is the full content of the file:

```
// Copyright 2011 Google Inc. All Rights Reserved.
//
// Use of this source code is governed by a BSD-style license
// that can be found in the COPYING file in the root of the source
// tree. An additional intellectual property rights grant can be found
// in the file PATENTS. All contributing project authors may
// be found in the AUTHORS file in the root of the source tree.
// -----------------------------------------------------------------------------
//
//  RIFF container manipulation for WebP images.
//
// Authors: Urvang (urvang@google.com)
//          Vikas (vikasa@google.com)

// This API allows manipulation of WebP container images containing features
// like color profile, metadata, animation and fragmented images.
//
// Code Example#1: Create a WebPMux object with image data, color profile and
// XMP metadata.
/*
  int copy_data = 0;
  WebPMux* mux = WebPMuxNew();
  // ... (Prepare image data).
  WebPMuxSetImage(mux, &image, copy_data);
  // ... (Prepare ICCP color profile data).
  WebPMuxSetChunk(mux, "ICCP", &icc_profile, copy_data);
  // ... (Prepare XMP metadata).
  WebPMuxSetChunk(mux, "XMP ", &xmp, copy_data);
  // Get data from mux in WebP RIFF format.
  WebPMuxAssemble(mux, &output_data);
  WebPMuxDelete(mux);
  // ... (Consume output_data; e.g. write output_data.bytes to file).
  WebPDataClear(&output_data);
*/

// Code Example#2: Get image and color profile data from a WebP file.
/*
  int copy_data = 0;
  // ... (Read data from file).
  WebPMux* mux = WebPMuxCreate(&data, copy_data);
  WebPMuxGetFrame(mux, 1, &image);
  // ... (Consume image; e.g. call WebPDecode() to decode the data).
  WebPMuxGetChunk(mux, "ICCP", &icc_profile);
  // ... (Consume icc_data).
  WebPMuxDelete(mux);
  free(data);
*/

#ifndef WEBP_WEBP_MUX_H_
#define WEBP_WEBP_MUX_H_

#include "./mux_types.h"

#ifdef __cplusplus
extern "C" {
#endif

#define WEBP_MUX_ABI_VERSION 0x0101        // MAJOR(8b) + MINOR(8b)

// Note: forward declaring enumerations is not allowed in (strict) C and C++,
// the types are left here for reference.
// typedef enum WebPMuxError WebPMuxError;
// typedef enum WebPChunkId WebPChunkId;
typedef struct WebPMux WebPMux;   // main opaque object.
typedef struct WebPMuxFrameInfo WebPMuxFrameInfo;
typedef struct WebPMuxAnimParams WebPMuxAnimParams;

// Error codes
typedef enum WebPMuxError {
  WEBP_MUX_OK                 =  1,
  WEBP_MUX_NOT_FOUND          =  0,
  WEBP_MUX_INVALID_ARGUMENT   = -1,
  WEBP_MUX_BAD_DATA           = -2,
  WEBP_MUX_MEMORY_ERROR       = -3,
  WEBP_MUX_NOT_ENOUGH_DATA    = -4
} WebPMuxError;

// IDs for different types of chunks.
typedef enum WebPChunkId {
  WEBP_CHUNK_VP8X,     // VP8X
  WEBP_CHUNK_ICCP,     // ICCP
  WEBP_CHUNK_ANIM,     // ANIM
  WEBP_CHUNK_ANMF,     // ANMF
  WEBP_CHUNK_FRGM,     // FRGM
  WEBP_CHUNK_ALPHA,    // ALPH
  WEBP_CHUNK_IMAGE,    // VP8/VP8L
  WEBP_CHUNK_EXIF,     // EXIF
  WEBP_CHUNK_XMP,      // XMP
  WEBP_CHUNK_UNKNOWN,  // Other chunks.
  WEBP_CHUNK_NIL
} WebPChunkId;

//------------------------------------------------------------------------------

// Returns the version number of the mux library, packed in hexadecimal using
// 8bits for each of major/minor/revision. E.g: v2.5.7 is 0x020507.
WEBP_EXTERN(int) WebPGetMuxVersion(void);

//------------------------------------------------------------------------------
// Life of a Mux object

// Internal, version-checked, entry point
WEBP_EXTERN(WebPMux*) WebPNewInternal(int);

// Creates an empty mux object.
// Returns:
//   A pointer to the newly created empty mux object.
//   Or NULL in case of memory error.
static WEBP_INLINE WebPMux* WebPMuxNew(void) {
  return WebPNewInternal(WEBP_MUX_ABI_VERSION);
}

// Deletes the mux object.
// Parameters:
//   mux - (in/out) object to be deleted
WEBP_EXTERN(void) WebPMuxDelete(WebPMux* mux);

//------------------------------------------------------------------------------
// Mux creation.

// Internal, version-checked, entry point
WEBP_EXTERN(WebPMux*) WebPMuxCreateInternal(const WebPData*, int, int);

// Creates a mux object from raw data given in WebP RIFF format.
// Parameters:
//   bitstream - (in) the bitstream data in WebP RIFF format
//   copy_data - (in) value 1 indicates given data WILL be copied to the mux
//               object and value 0 indicates data will NOT be copied.
// Returns:
//   A pointer to the mux object created from given data - on success.
//   NULL - In case of invalid data or memory error.
static WEBP_INLINE WebPMux* WebPMuxCreate(const WebPData* bitstream,
                                          int copy_data) {
  return WebPMuxCreateInternal(bitstream, copy_data, WEBP_MUX_ABI_VERSION);
}

//------------------------------------------------------------------------------
// Non-image chunks.

// Note: Only non-image related chunks should be managed through chunk APIs.
// (Image related chunks are: "ANMF", "FRGM", "VP8 ", "VP8L" and "ALPH").
// To add, get and delete images, use WebPMuxSetImage(), WebPMuxPushFrame(),
// WebPMuxGetFrame() and WebPMuxDeleteFrame().

// Adds a chunk with id 'fourcc' and data 'chunk_data' in the mux object.
// Any existing chunk(s) with the same id will be removed.
// Parameters:
//   mux - (in/out) object to which the chunk is to be added
//   fourcc - (in) a character array containing the fourcc of the given chunk;
//                 e.g., "ICCP", "XMP ", "EXIF" etc.
//   chunk_data - (in) the chunk data to be added
//   copy_data - (in) value 1 indicates given data WILL be copied to the mux
//               object and value 0 indicates data will NOT be copied.
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux, fourcc or chunk_data is NULL
//                               or if fourcc corresponds to an image chunk.
//   WEBP_MUX_MEMORY_ERROR - on memory allocation error.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxSetChunk(
    WebPMux* mux, const char fourcc[4], const WebPData* chunk_data,
    int copy_data);

// Gets a reference to the data of the chunk with id 'fourcc' in the mux object.
// The caller should NOT free the returned data.
// Parameters:
//   mux - (in) object from which the chunk data is to be fetched
//   fourcc - (in) a character array containing the fourcc of the chunk;
//                 e.g., "ICCP", "XMP ", "EXIF" etc.
//   chunk_data - (out) returned chunk data
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux, fourcc or chunk_data is NULL
//                               or if fourcc corresponds to an image chunk.
//   WEBP_MUX_NOT_FOUND - If mux does not contain a chunk with the given id.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxGetChunk(
    const WebPMux* mux, const char fourcc[4], WebPData* chunk_data);

// Deletes the chunk with the given 'fourcc' from the mux object.
// Parameters:
//   mux - (in/out) object from which the chunk is to be deleted
//   fourcc - (in) a character array containing the fourcc of the chunk;
//                 e.g., "ICCP", "XMP ", "EXIF" etc.
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux or fourcc is NULL
//                               or if fourcc corresponds to an image chunk.
//   WEBP_MUX_NOT_FOUND - If mux does not contain a chunk with the given fourcc.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxDeleteChunk(
    WebPMux* mux, const char fourcc[4]);

//------------------------------------------------------------------------------
// Images.

// Encapsulates data about a single frame/fragment.
struct WebPMuxFrameInfo {
  WebPData    bitstream;  // image data: can be a raw VP8/VP8L bitstream
                          // or a single-image WebP file.
  int         x_offset;   // x-offset of the frame.
  int         y_offset;   // y-offset of the frame.
  int         duration;   // duration of the frame (in milliseconds).

  WebPChunkId id;         // frame type: should be one of WEBP_CHUNK_ANMF,
                          // WEBP_CHUNK_FRGM or WEBP_CHUNK_IMAGE
  WebPMuxAnimDispose dispose_method;  // Disposal method for the frame.
  WebPMuxAnimBlend   blend_method;    // Blend operation for the frame.
  uint32_t    pad[1];     // padding for later use
};

// Sets the (non-animated and non-fragmented) image in the mux object.
// Note: Any existing images (including frames/fragments) will be removed.
// Parameters:
//   mux - (in/out) object in which the image is to be set
//   bitstream - (in) can be a raw VP8/VP8L bitstream or a single-image
//               WebP file (non-animated and non-fragmented)
//   copy_data - (in) value 1 indicates given data WILL be copied to the mux
//               object and value 0 indicates data will NOT be copied.
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux is NULL or bitstream is NULL.
//   WEBP_MUX_MEMORY_ERROR - on memory allocation error.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxSetImage(
    WebPMux* mux, const WebPData* bitstream, int copy_data);

// Adds a frame at the end of the mux object.
// Notes: (1) frame.id should be one of WEBP_CHUNK_ANMF or WEBP_CHUNK_FRGM
//        (2) For setting a non-animated non-fragmented image, use
//            WebPMuxSetImage() instead.
//        (3) Type of frame being pushed must be same as the frames in mux.
//        (4) As WebP only supports even offsets, any odd offset will be snapped
//            to an even location using: offset &= ~1
// Parameters:
//   mux - (in/out) object to which the frame is to be added
//   frame - (in) frame data.
//   copy_data - (in) value 1 indicates given data WILL be copied to the mux
//               object and value 0 indicates data will NOT be copied.
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux or frame is NULL
//                               or if content of 'frame' is invalid.
//   WEBP_MUX_MEMORY_ERROR - on memory allocation error.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxPushFrame(
    WebPMux* mux, const WebPMuxFrameInfo* frame, int copy_data);

// Gets the nth frame from the mux object.
// The content of 'frame->bitstream' is allocated using malloc(), and NOT
// owned by the 'mux' object. It MUST be deallocated by the caller by calling
// WebPDataClear().
// nth=0 has a special meaning - last position.
// Parameters:
//   mux - (in) object from which the info is to be fetched
//   nth - (in) index of the frame in the mux object
//   frame - (out) data of the returned frame
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux or frame is NULL.
//   WEBP_MUX_NOT_FOUND - if there are less than nth frames in the mux object.
//   WEBP_MUX_BAD_DATA - if nth frame chunk in mux is invalid.
//   WEBP_MUX_MEMORY_ERROR - on memory allocation error.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxGetFrame(
    const WebPMux* mux, uint32_t nth, WebPMuxFrameInfo* frame);

// Deletes a frame from the mux object.
// nth=0 has a special meaning - last position.
// Parameters:
//   mux - (in/out) object from which a frame is to be deleted
//   nth - (in) The position from which the frame is to be deleted
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux is NULL.
//   WEBP_MUX_NOT_FOUND - If there are less than nth frames in the mux object
//                        before deletion.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxDeleteFrame(WebPMux* mux, uint32_t nth);

//------------------------------------------------------------------------------
// Animation.

// Animation parameters.
struct WebPMuxAnimParams {
  uint32_t bgcolor;  // Background color of the canvas stored (in MSB order) as:
                     // Bits 00 to 07: Alpha.
                     // Bits 08 to 15: Red.
                     // Bits 16 to 23: Green.
                     // Bits 24 to 31: Blue.
  int loop_count;    // Number of times to repeat the animation [0 = infinite].
};

// Sets the animation parameters in the mux object. Any existing ANIM chunks
// will be removed.
// Parameters:
//   mux - (in/out) object in which ANIM chunk is to be set/added
//   params - (in) animation parameters.
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux or params is NULL.
//   WEBP_MUX_MEMORY_ERROR - on memory allocation error.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxSetAnimationParams(
    WebPMux* mux, const WebPMuxAnimParams* params);

// Gets the animation parameters from the mux object.
// Parameters:
//   mux - (in) object from which the animation parameters to be fetched
//   params - (out) animation parameters extracted from the ANIM chunk
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux or params is NULL.
//   WEBP_MUX_NOT_FOUND - if ANIM chunk is not present in mux object.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxGetAnimationParams(
    const WebPMux* mux, WebPMuxAnimParams* params);

//------------------------------------------------------------------------------
// Misc Utilities.

#if WEBP_MUX_ABI_VERSION > 0x0101
// Sets the canvas size for the mux object. The width and height can be
// specified explicitly or left as zero (0, 0).
// * When width and height are specified explicitly, then this frame bound is
//   enforced during subsequent calls to WebPMuxAssemble() and an error is
//   reported if any animated frame does not completely fit within the canvas.
// * When unspecified (0, 0), the constructed canvas will get the frame bounds
//   from the bounding-box over all frames after calling WebPMuxAssemble().
// Parameters:
//   mux - (in) object to which the canvas size is to be set
//   width - (in) canvas width
//   height - (in) canvas height
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux is NULL; or
//                               width or height are invalid or out of bounds
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxSetCanvasSize(WebPMux* mux,
                                               int width, int height);
#endif

// Gets the canvas size from the mux object.
// Note: This method assumes that the VP8X chunk, if present, is up-to-date.
// That is, the mux object hasn't been modified since the last call to
// WebPMuxAssemble() or WebPMuxCreate().
// Parameters:
//   mux - (in) object from which the canvas size is to be fetched
//   width - (out) canvas width
//   height - (out) canvas height
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux, width or height is NULL.
//   WEBP_MUX_BAD_DATA - if VP8X/VP8/VP8L chunk or canvas size is invalid.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxGetCanvasSize(const WebPMux* mux,
                                               int* width, int* height);

// Gets the feature flags from the mux object.
// Note: This method assumes that the VP8X chunk, if present, is up-to-date.
// That is, the mux object hasn't been modified since the last call to
// WebPMuxAssemble() or WebPMuxCreate().
// Parameters:
//   mux - (in) object from which the features are to be fetched
//   flags - (out) the flags specifying which features are present in the
//           mux object. This will be an OR of various flag values.
//           Enum 'WebPFeatureFlags' can be used to test individual flag values.
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux or flags is NULL.
//   WEBP_MUX_BAD_DATA - if VP8X/VP8/VP8L chunk or canvas size is invalid.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxGetFeatures(const WebPMux* mux,
                                             uint32_t* flags);

// Gets number of chunks with the given 'id' in the mux object.
// Parameters:
//   mux - (in) object from which the info is to be fetched
//   id - (in) chunk id specifying the type of chunk
//   num_elements - (out) number of chunks with the given chunk id
// Returns:
//   WEBP_MUX_INVALID_ARGUMENT - if mux, or num_elements is NULL.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxNumChunks(const WebPMux* mux,
                                           WebPChunkId id, int* num_elements);

// Assembles all chunks in WebP RIFF format and returns in 'assembled_data'.
// This function also validates the mux object.
// Note: The content of 'assembled_data' will be ignored and overwritten.
// Also, the content of 'assembled_data' is allocated using malloc(), and NOT
// owned by the 'mux' object. It MUST be deallocated by the caller by calling
// WebPDataClear(). It's always safe to call WebPDataClear() upon return,
// even in case of error.
// Parameters:
//   mux - (in/out) object whose chunks are to be assembled
//   assembled_data - (out) assembled WebP data
// Returns:
//   WEBP_MUX_BAD_DATA - if mux object is invalid.
//   WEBP_MUX_INVALID_ARGUMENT - if mux or assembled_data is NULL.
//   WEBP_MUX_MEMORY_ERROR - on memory allocation error.
//   WEBP_MUX_OK - on success.
WEBP_EXTERN(WebPMuxError) WebPMuxAssemble(WebPMux* mux,
                                          WebPData* assembled_data);

//------------------------------------------------------------------------------

#ifdef __cplusplus
}    // extern "C"
#endif

#endif  /* WEBP_WEBP_MUX_H_ */
```

**Comment by Abdelhady** [comment]
Any update on this one?

**Comment by dlemstra** [comment]
I think this was resolved with the following patch: 2fcb93734b2030bc4185d6cab275c607339084e3. We changed the check from `0.4.4` to `0.5.0`. Can you give it a try with the latest release?

**Comment by Abdelhady** [comment]
Great, it is working fine now :+1:, with WebP `0.4.4-1` (latest on ubuntu) & latest ImageMagick `7.0.9-2`

**Comment by Stafox** [comment]
@dlemstra hi.
got the same case as @Abdelhady . And got absolutely the same error trying build ImageMagick 7.0.10-34 with webp support.

```
identify -list format| grep webp
     WEBP* rw-   WebP Image Format (libwebp 0.6.0 [020E])
```

```
  CC       coders/MagickCore_libMagickCore_7_Q16HDRI_la-png.lo
coders/png.c: In function ‘ReadOnePNGImage’:
coders/png.c:2453:8: warning: unused variable ‘option’ [-Wunused-variable]
       *option;
        ^
  CC       coders/MagickCore_libMagickCore_7_Q16HDRI_la-ept.lo
  CC       coders/MagickCore_libMagickCore_7_Q16HDRI_la-tiff.lo
  CC       coders/MagickCore_libMagickCore_7_Q16HDRI_la-webp.lo
In file included from coders/webp.c:75:0:
/usr/include/webp/mux.h:101:13: error: expected identifier or ‘(’ before ‘int’
 WEBP_EXTERN(int) WebPGetMuxVersion(void);
             ^
/usr/include/webp/mux.h:107:13: error: expected identifier or ‘(’ before ‘WebPMux’
 WEBP_EXTERN(WebPMux*) WebPNewInternal(int);
             ^
/usr/include/webp/mux.h:120:13: error: expected identifier or ‘(’ before ‘void’
 WEBP_EXTERN(void) WebPMuxDelete(WebPMux* mux);
             ^
/usr/include/webp/mux.h:126:13: error: expected identifier or ‘(’ before ‘WebPMux’
 WEBP_EXTERN(WebPMux*) WebPMuxCreateInternal(const WebPData*, int, int);
             ^
/usr/include/webp/mux.h:163:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxSetChunk(
             ^
/usr/include/webp/mux.h:179:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxGetChunk(
             ^
/usr/include/webp/mux.h:192:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxDeleteChunk(
             ^
/usr/include/webp/mux.h:225:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxSetImage(
             ^
/usr/include/webp/mux.h:244:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxPushFrame(
             ^
/usr/include/webp/mux.h:262:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxGetFrame(
             ^
/usr/include/webp/mux.h:275:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxDeleteFrame(WebPMux* mux, uint32_t nth);
             ^
/usr/include/webp/mux.h:299:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxSetAnimationParams(
             ^
/usr/include/webp/mux.h:310:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxGetAnimationParams(
             ^
/usr/include/webp/mux.h:331:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxSetCanvasSize(WebPMux* mux,
             ^
/usr/include/webp/mux.h:346:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxGetCanvasSize(const WebPMux* mux,
             ^
/usr/include/webp/mux.h:362:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxGetFeatures(const WebPMux* mux,
             ^
/usr/include/webp/mux.h:373:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxNumChunks(const WebPMux* mux,
             ^
/usr/include/webp/mux.h:391:13: error: expected identifier or ‘(’ before ‘WebPMuxError’
 WEBP_EXTERN(WebPMuxError) WebPMuxAssemble(WebPMux* mux,
             ^
/usr/include/webp/mux.h:445:13: error: expected identifier or ‘(’ before ‘int’
 WEBP_EXTERN(int) WebPAnimEncoderOptionsInitInternal(
             ^
/usr/include/webp/mux.h:458:13: error: expected identifier or ‘(’ before ‘WebPAnimEncoder’
 WEBP_EXTERN(WebPAnimEncoder*) WebPAnimEncoderNewInternal(
             ^
/usr/include/webp/mux.h:493:13: error: expected identifier or ‘(’ before ‘int’
 WEBP_EXTERN(int) WebPAnimEncoderAdd(
             ^
/usr/include/webp/mux.h:506:13: error: expected identifier or ‘(’ before ‘int’
 WEBP_EXTERN(int) WebPAnimEncoderAssemble(WebPAnimEncoder* enc,
             ^
/usr/include/webp/mux.h:517:13: error: expected identifier or ‘(’ before ‘const’
 WEBP_EXTERN(const char*) WebPAnimEncoderGetError(WebPAnimEncoder* enc);
             ^
/usr/include/webp/mux.h:522:13: error: expected identifier or ‘(’ before ‘void’
 WEBP_EXTERN(void) WebPAnimEncoderDelete(WebPAnimEncoder* enc);
             ^
coders/webp.c: In function ‘ReadSingleWEBPImage’:
coders/webp.c:358:5: warning: implicit declaration of function ‘WebPMuxGetFeatures’ [-Wimplicit-function-declaration]
     WebPMuxGetFeatures(mux,&webp_flags);
     ^
coders/webp.c:361:9: warning: implicit declaration of function ‘WebPMuxGetChunk’ [-Wimplicit-function-declaration]
         WebPMuxGetChunk(mux,"ICCP",&chunk);
         ^
coders/webp.c:389:5: warning: implicit declaration of function ‘WebPMuxDelete’ [-Wimplicit-function-declaration]
     WebPMuxDelete(mux);
     ^
coders/webp.c: In function ‘WriteAnimatedWEBPImage’:
coders/webp.c:879:5: warning: implicit declaration of function ‘WebPAnimEncoderAdd’ [-Wimplicit-function-declaration]
     WebPAnimEncoderAdd(enc,&picture,(int) frame_timestamp,configure);
     ^
coders/webp.c:887:3: warning: implicit declaration of function ‘WebPAnimEncoderAssemble’ [-Wimplicit-function-declaration]
   WebPAnimEncoderAssemble(enc, &webp_data);
   ^
coders/webp.c:891:3: warning: implicit declaration of function ‘WebPAnimEncoderDelete’ [-Wimplicit-function-declaration]
   WebPAnimEncoderDelete(enc);
   ^
coders/webp.c: In function ‘WriteWEBPImage’:
coders/webp.c:1156:19: warning: implicit declaration of function ‘WebPMuxSetChunk’ [-Wimplicit-function-declaration]
         mux_error=WebPMuxSetChunk(mux,"ICCP",&chunk,0);
                   ^
coders/webp.c:1185:9: warning: implicit declaration of function ‘WebPMuxSetImage’ [-Wimplicit-function-declaration]
         WebPMuxSetImage(mux,&image_chunk,1);
         ^
coders/webp.c:1186:19: warning: implicit declaration of function ‘WebPMuxAssemble’ [-Wimplicit-function-declaration]
         mux_error=WebPMuxAssemble(mux,&picture_profiles);
                   ^
Makefile:10010: recipe for target 'coders/MagickCore_libMagickCore_7_Q16HDRI_la-webp.lo' failed
make[1]: *** [coders/MagickCore_libMagickCore_7_Q16HDRI_la-webp.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/tmp/ImageMagick-7.0.10-34'
Makefile:6033: recipe for target 'all' failed
make: *** [all] Error 2
```

WebPMux package on my server:
```
libwebpmux3/xenial,now 0.6.0-4+ubuntu16.04.1+deb.sury.org+1 amd64 [installed,automatic]
```

Do you have any idea why this happens and how to fix it?
